sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -241,6 +246,11 @@ protected String visitSQLFunction(String funcName, String[] inputs) {
     return funcName + "(" + Arrays.stream(inputs).collect(Collectors.joining(", ")) + ")";
   }
 
+  protected String visitUserDefinedFunction(
+      String funcName, String canonicalName, String[] inputs) {
+    return funcName + "(" + Arrays.stream(inputs).collect(Collectors.joining(", ")) + ")";
```

Comment:
I think it's safer to fail by default here. We must ask the data sources to carefully check the `canonicalName` to make sure they support the UDF.